### PR TITLE
`oauth_signature` not encoded in versions 1.2.0 and 1.3.0

### PR DESCRIPTION
**Pull request by ech0s7r**
### PR checklist

- [x] An issue/feature request has been created for this PR
- [x] Pull Request title clearly describes the work in the pull request and the Pull Request description provides details about how to validate the work. Missing information here may result in a delayed response.
- [x] File the PR against the `main` branch
- [x] The code in this PR is covered by unit tests

#### Description
This PR contains the following changes: 
- Code cleanup (format, indentation, ...) 
- Fix issue https://github.com/Mastercard/oauth1-signer-python/issues/35 (revert change introduced with https://github.com/Mastercard/oauth1-signer-python/pull/30)
- Fixed security hotspot [python:S2245](https://rules.sonarsource.com/python/RSPEC-2245)
- Keep the behavior consistent with other signer libs when encoding URL and params (URL and params should be percent-encoded and only query string should be encoded during params normalization)
- The OAuth1RSA module reuse the core functions instead of redefining them 
- `sha256_encode`: added support for `byte` payload
- The `oauth_signature` param in the Authorization header should be RFC-3986 encoded
- Moved `nonce` and `timestamp` generation function into the utils module 
- `get_authorization_header` method is now static 
- Improved code coverage 
- Added pycodestyle GitHub workflow and fixed sonar scan not running on pull requests

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  schedule:
+    - cron: 0 16 * * *
+jobs:
+  sonarcloud:
+    name: Lint
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Python Style Checker
+        uses: andymckay/pycodestyle-action@0.1.3
```
Comment (jaaufauvre) [inline .github/workflows/lint.yml:18]: 👍

File: .github/workflows/sonar.yml [hunk 2]
```diff
@@ -2,12 +2,10 @@ name: Sonar
 'on':
   push:
     branches:
-      - main
-  pull_request_target:
-    types:
-      - opened
-      - synchronize
-      - reopened
+      - master
```
Comment (jaaufauvre) [inline .github/workflows/sonar.yml]: This should be `main` now that we renamed `master` into `main` (or the workflows won't run)

File: oauth1/signer_interceptor.py [hunk 3]
```diff
@@ -1,3 +1,30 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2019-2020 Mastercard
```
Comment (jaaufauvre) [inline oauth1/signer_interceptor.py]: Minor, but maybe use "2021" (in all headers 😅)

**Review by jaaufauvre — CHANGES_REQUESTED** [review]
Thanks @ech0s7r! I left some feedback (easy to fix)

**Comment by ech0s7r** [comment]
> Thanks @ech0s7r! I left some feedback (easy to fix)

Thank you, @jaaufauvre for the review! :) 

Updated the Copyright header (`2020` -> `2021`) and branch name (`master` -> `main`) in 0bac0f88. 

image

**Review by jaaufauvre — APPROVED** [review]

**Pull request merged**